src/v/cloud_storage/tests/segment_meta_cstore_test.cc

```diff
@@ -1107,3 +1113,116 @@ BOOST_AUTO_TEST_CASE(test_segment_meta_cstore_insert_replacements) {
       merged_result.begin(),
       merged_result.end()));
 }
+
+BOOST_AUTO_TEST_CASE(
+  test_deltafor_generate_characterization_data, *boost::unit_test::disabled()) {
+    /** this test is disabled because it generates a bunch of data to be used
+     * for characterization of deltafor, and prints it to stdout ready to paste
+     * in a file and use it for unit tests*/
+
+    struct datum {
+        std::array<int64_t, details::FOR_buffer_depth> ref;
+        std::vector<uint8_t> bytes;
+    };
+
+    auto to_save = std::vector<datum>{};
+
+    // generate a datum for each N_BITS [0, 64], with the bounds being the edge
+    // cases
+    for (auto i = 0u; i <= sizeof(int64_t) * 8; ++i) {
+        auto dfor = deltafor_datapoint::dfor_enc{};
+        auto buffer = std::array<int64_t, details::FOR_buffer_depth>{};
+        if (i > 0) {
+            // random bits
+            std::ranges::generate(buffer, [] {
+                return random_generators::get_int(
+                  std::numeric_limits<int64_t>::min(),
+                  std::numeric_limits<int64_t>::max());
+            });
+
+            // clear out upper bits, and set an odd number of msb to 1 to
+            // ensure that delta_xor produces N_BITS=i
+            auto keep_mask = ~(std::numeric_limits<uint64_t>::max() << (i - 1));
+            for (auto& e : buffer) {
+                e &= keep_mask;
+            }
+            buffer[2] |= uint64_t{1} << (i - 1);
+            buffer[5] |= uint64_t{1} << (i - 1);
+            buffer[11] |= uint64_t{1} << (i - 1);
+        }
+        dfor.add(buffer);
+        auto serialized = iobuf_parser{serde::to_iobuf(std::move(dfor))};
+        auto tmp = std::vector<uint8_t>(serialized.bytes_left(), 0);
+        serialized.consume_to(serialized.bytes_left(), tmp.begin());
+        to_save.emplace_back(buffer, std::move(tmp));
+    }
+
+    auto deltafor_datapoint_printer = [](auto& p) {
+        return fmt::format(
+          "{{ (int64_t[]) {{ {} }},\n(uint8_t[]) {{ {} }} }}",
+          fmt::join(p.ref, ","),
+          fmt::join(p.bytes, ","));
+    };
+    fmt::print(
+      R"cpp(
+
+#include "delta_for_characterization_data.h"
+
+#include <cstdint>
+
+// data produced from {}
+
+constexpr auto characterization_data = std::to_array<deltafor_datapoint>({{
+{}
+}});
+
+auto get_characterization_data() -> std::span<const deltafor_datapoint> {{
+    return characterization_data;
+}}
+)cpp",
+      redpanda_version(),
+      fmt::join(
+        to_save | std::views::transform(deltafor_datapoint_printer), ",\n"));
+}
+
+BOOST_AUTO_TEST_CASE(test_deltafor_characterization) {
```

Comment:
this should probably live inside the util/tests in delta_for test suite

Reply:
right, makes more sense there